packages/@aws-cdk/aws-sagemaker/test/integ.endpoint.ts

```diff
@@ -0,0 +1,139 @@
+import * as path from 'path';
+import * as iam from '@aws-cdk/aws-iam';
+import * as cdk from '@aws-cdk/core';
+import { ExpectedResult, IntegTest } from '@aws-cdk/integ-tests';
+import * as sagemaker from '../lib';
+
+/*
+ * Stack verification steps:
+ * aws sagemaker-runtime invoke-endpoint --endpoint-name <deployed endpoint name> --body "any string" --cli-binary-format raw-in-base64-out /tmp/inference.txt && cat /tmp/inference.txt
+ *
+ * The above command will result in one of the following outputs (based on relative variant weight).
+ *
+ * Roughly 25% of the time:
+ *
+ *   {
+ *       "ContentType": "text/plain",
+ *       "InvokedProductionVariant": "firstVariant"
+ *   }
+ *   Artifact info: ['This file exists for test purposes only and is not a real SageMaker Model artifact']
+ *
+ * Roughly 25% of the time:
+ *
+ *   {
+ *       "ContentType": "text/plain",
+ *       "InvokedProductionVariant": "secondVariant"
+ *   }
+ *   Artifact info: ['This file exists for test purposes only and is not a real SageMaker Model artifact']
+ *
+ * Roughly 50% of the time:
+ *
+ *   {
+ *       "ContentType": "text/plain",
+ *       "InvokedProductionVariant": "thirdVariant"
+ *   }
+ *   Artifact info: No artifacts are present
+ */
+
+const app = new cdk.App();
+const stack = new cdk.Stack(app, 'aws-cdk-sagemaker-endpoint');
+
+const image = sagemaker.ContainerImage.fromAsset(path.join(__dirname, 'test-image'));
+const modelData = sagemaker.ModelData.fromAsset(path.join(__dirname, 'test-artifacts', 'valid-artifact.tar.gz'));
+
+const modelWithArtifact = new sagemaker.Model(stack, 'ModelWithArtifact', {
+  containers: [{ image, modelData }],
+});
+const modelWithoutArtifact = new sagemaker.Model(stack, 'ModelWithoutArtifact', {
+  containers: [{ image }],
+});
+
+const endpointConfig = new sagemaker.EndpointConfig(stack, 'EndpointConfig', {
+  instanceProductionVariants: [
+    {
+      model: modelWithArtifact,
+      variantName: 'firstVariant',
+      instanceType: sagemaker.InstanceType.M5_LARGE,
+    },
+    {
+      model: modelWithArtifact,
+      variantName: 'secondVariant',
+    },
+  ],
+});
+endpointConfig.addInstanceProductionVariant({
+  model: modelWithoutArtifact,
+  variantName: 'thirdVariant',
+  initialVariantWeight: 2.0,
+});
+const endpoint = new sagemaker.Endpoint(stack, 'Endpoint', { endpointConfig });
+
+const invokerRole = new iam.Role(stack, 'Invoker', {
+  assumedBy: new iam.AccountRootPrincipal(),
+});
+endpoint.grantInvoke(invokerRole);
+
+const productionVariant = endpoint.findInstanceProductionVariant('firstVariant');
+const instanceCount = productionVariant.autoScaleInstanceCount({
+  maxCapacity: 3,
+});
+instanceCount.scaleOnInvocations('LimitRPS', {
+  maxRequestsPerSecond: 30,
+});
+
+productionVariant.metricInvocations().createAlarm(stack, 'InvocationsAlarm', {
+  threshold: 1,
+  evaluationPeriods: 2,
+});
```

Comment:
last thing... these alarms don't seem to be needed since we're not asserting that they exist or anything. Lets move this to a separate integ test as it doesn't really test the endpoint construct. maybe `integ.alarms.ts` or something.

It is cluttering the current integ test atm.